src/common/snippets/src/lowered/pass/pass.cpp

```diff
@@ -31,7 +36,35 @@ void PassPipeline::run(LinearIR& linear_ir) const {
 }
 
 void PassPipeline::run(LinearIR& linear_ir, LinearIR::constExprIt begin, LinearIR::constExprIt end) const {
+#ifdef SNIPPETS_DEBUG_CAPS
+    const auto& dumpLIR = linear_ir.get_config().debug_config.dumpLIR;
+    bool enable_dump = false;
+    std::string pass_key;
+    std::string directory;
+    if (!dumpLIR.passes.empty()) {
+        enable_dump = true;
+        pass_key = dumpLIR.passes;
+        directory = dumpLIR.dir + "/";
+    }
+#endif
     for (const auto& pass : m_passes) {
+#ifdef SNIPPETS_DEBUG_CAPS
+        bool actived_pass = false;
+        auto pass_name = std::string(pass->get_type_name());
+        if (enable_dump && (pass_key == "all" || (pass_name.find(pass_key) != std::string::npos))) {
+            actived_pass = true;
+            if (dumpLIR.format.filter[DebugCapsConfig::LIRFormatFilter::controlFlow]) {
+                std::string xml_path = directory + pass_name + "_control_flow_in.xml";
+                lowered::pass::SerializeControlFlow SerializeLIR(xml_path);
+                SerializeLIR.run(linear_ir);
+            }
+            if (dumpLIR.format.filter[DebugCapsConfig::LIRFormatFilter::dataFlow]) {
+                std::string xml_path = directory + pass_name + "_data_flow_in.xml";
+                lowered::pass::SerializeDataFlow SerializeLIR(xml_path);
+                SerializeLIR.run(linear_ir);
+            }
+        }
+#endif
```

Comment:
I'd prefer to encapsulate debug caps related logic in a separate entity, and leave here only some one-line call (or maybe macros?). Could you please take a look at [TransformationDumper](https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/utils/ngraph_transformation.hpp) class and take it as an example?

Reply:
great suggestion, have done. Thanks Vladislav!